sql/linestring.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020-2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/dolthub/vitess/go/sqltypes"
+	"github.com/dolthub/vitess/go/vt/proto/query"
+)
+
+// Represents the Linestring type.
+// https://dev.mysql.com/doc/refman/8.0/en/gis-class-point.html
+type LinestringType interface {
+	Type
+}
+
+type LinestringValue struct {
+	Points []PointValue
+}
+
+var Linestring LinestringType = LinestringValue{}
+
+// Compare implements Type interface.
+func (t LinestringValue) Compare(a interface{}, b interface{}) (int, error) {
+	// Compare nulls
+	if hasNulls, res := compareNulls(a, b); hasNulls {
+		return res, nil
+	}
+
+	// Cast to linestring
+	_a, err := t.convertToLinestringValue(a)
+	if err != nil {
+		return 0, err
+	}
+	_b, err := t.convertToLinestringValue(b)
+	if err != nil {
+		return 0, err
+	}
+
+	// Get shorter length
+	var n int
+	lenA := len(_a.Points)
+	lenB := len(_b.Points)
+	if lenA < lenB {
+		n = lenA
+	} else {
+		n = lenB
+	}
+
+	// Compare each point until there's a difference
+	for i := 0; i < n; i++ {
+		diff, err := PointValue{}.Compare(_a.Points[i], _b.Points[i])
+		if err != nil {
+			return 0, err
+		}
+		if diff != 0 {
+			return diff, nil
+		}
+	}
+
+	// Determine based off length
+	if lenA > lenB {
+		return 1, nil
+	}
+	if lenA < lenB {
+		return -1, nil
+	}
+
+	return 0, nil
+}
+
+func (t LinestringValue) convertToLinestringValue(v interface{}) (LinestringValue, error) {
+	switch v := v.(type) {
+	case LinestringValue:
+		return v, nil
+	case string:
+		// TODO: janky parsing
```

Comment:
I don't think you want to do this. From what I can tell, MySQL does not coerce strings to geometry value for this funciton